R/calculate.R

```diff
@@ -164,6 +153,60 @@ calculate <- function(x,
   result
 }
 
+# Raise an error if the user supplies a test statistic that doesn't
+# make sense given the variable specified
+check_variables_vs_stat <- function(x, stat) {
+  if (!stat %in% implemented_stats) {
+    stop_glue(
+      "You specified a string for `stat` that is not implemented. ",
+      "Check your spelling and `?calculate` for current options."
+    )
+  }
+  
+  res_type <- determine_variable_type(x, "response")
+  exp_type <- determine_variable_type(x, "explanatory")
+  
+  possible_stats <- stat_types %>%
+    dplyr::filter(resp == res_type & exp == exp_type) %>%
+    dplyr::pull(stats) %>%
+    unlist()
+
+  if (is.null(possible_stats)) {
+    stop_glue(
+      "The infer team has not implemented test statistics for the ",
+      "supplied variable types."
+    )
+  }
+  
+  if (!stat %in% possible_stats) {
+    stop_glue(
+      stat_desc$description[stat_desc$stat == stat],
```

Comment:
I couldn't match this condition anywhere else in the source. Where is it that you have in mind?

Reply:
Didn't clarify enough. I meant "looking up from a specified table" pattern: [here](https://github.com/tidymodels/infer/blob/4352ecc1433b66b499724da85809a47673c5d42c/R/calculate.R#L146), [here](https://github.com/tidymodels/infer/blob/4352ecc1433b66b499724da85809a47673c5d42c/R/calculate.R#L148-L150), [here](https://github.com/tidymodels/infer/blob/4352ecc1433b66b499724da85809a47673c5d42c/R/calculate.R#L153-L155), [here](https://github.com/tidymodels/infer/blob/4352ecc1433b66b499724da85809a47673c5d42c/R/calculate.R#L185), and [here](https://github.com/tidymodels/infer/blob/4352ecc1433b66b499724da85809a47673c5d42c/R/calculate.R#L204).
These seem to be repetitive. Maybe wrap those into `get_stat_desc(stat)` and `get_stat_type_desc(stat_type)`? It looks like a more "future-proof" version of achieving the same task of "get printable description of something". Maybe even better with single wrapper `get_description()`, but not sure if this will be an overkill.

Reply:
These `glue` calls are much more readable now, thanks!